program/tests/test_create_mint.rs

```diff
@@ -1,256 +1,103 @@
-const FREEZE_AUTHORITY: Pubkey =
-    Pubkey::from_str_const("11111115q4EpJaTXAZWpCg3J2zppWGSZ46KXozzo9");
-
-fn setup_mint(owner: Pubkey, rent: &Rent) -> Account {
-    let state = spl_token::state::Mint {
-        decimals: MINT_DECIMALS,
-        is_initialized: true,
-        supply: MINT_SUPPLY,
-        freeze_authority: COption::Some(FREEZE_AUTHORITY),
-        ..Default::default()
-    };
-    let mut data = vec![0u8; spl_token::state::Mint::LEN];
-    state.pack_into_slice(&mut data);
-
-    let lamports = rent.minimum_balance(data.len());
-
-    Account {
-        lamports,
-        data,
-        owner,
-        ..Default::default()
-    }
-}
+pub mod helpers;
 
 #[test]
 fn test_idempotency_false_with_existing_account() {
-    let program_id = spl_token_wrap::id();
-    let mut mollusk = Mollusk::new(&program_id, "spl_token_wrap");
-    mollusk_svm_programs_token::token2022::add_program(&mut mollusk);
-
-    let unwrapped_mint_address = Pubkey::new_unique();
-    let wrapped_token_program_id = spl_token_2022::id();
-    let wrapped_mint_address =
-        get_wrapped_mint_address(&unwrapped_mint_address, &wrapped_token_program_id);
-    let wrapped_backpointer_address = get_wrapped_mint_backpointer_address(&wrapped_mint_address);
-
-    let instruction = create_mint(
-        &program_id,
-        &wrapped_mint_address,
-        &wrapped_backpointer_address,
-        &unwrapped_mint_address,
-        &wrapped_token_program_id,
-        false,
-    );
-
-    // Simulating existing data on mint or backpointer
     let account_with_data = Account {
         data: vec![1; 10],
         ..Account::default()
     };
 
-    // idempotent: false causes these to fail
-    let accounts = &[
-        (wrapped_mint_address, account_with_data.clone()), // mint already exists
-        (wrapped_backpointer_address, Account::default()),
-        (unwrapped_mint_address, Account::default()),
-        (solana_system_interface::program::id(), Account::default()),
-        mollusk_svm_programs_token::token2022::keyed_account(),
-    ];
-    mollusk.process_and_validate_instruction(
-        &instruction,
-        accounts,
-        &[Check::err(ProgramError::AccountAlreadyInitialized)],
-    );
-
-    let accounts = &[
-        (wrapped_mint_address, Account::default()),
-        (wrapped_backpointer_address, account_with_data), // backpointer already exists
-        (unwrapped_mint_address, Account::default()),
-        (solana_system_interface::program::id(), Account::default()),
-        mollusk_svm_programs_token::token2022::keyed_account(),
-    ];
-    mollusk.process_and_validate_instruction(
-        &instruction,
-        accounts,
-        &[Check::err(ProgramError::AccountAlreadyInitialized)],
-    );
+    // Test case 1: mint already exists
+    MintBuilder::default()
+        .wrapped_mint_account(account_with_data.clone())
+        .check(Check::err(ProgramError::AccountAlreadyInitialized))
+        .execute();
```

Comment:
Wow, I never understood why people liked builder patterns so much, but this looks really nice!